lib/src/chart/line_chart/line_chart_data.dart

```diff
@@ -776,76 +776,169 @@ Color _defaultGetDotStrokeColor(FlSpot spot, double xPercentage, LineChartBarDat
-  final GetDotColorCallback getStrokeColor;
-
-  /// Callback which is called to the the color of the given [FlSpot].
-  /// The [FlSpot] is provided as parameter to this callback
-  final GetDotColorCallback getDotColor;
-
   /// Checks to show or hide an individual dot.
   final CheckToShowDot checkToShowDot;
 
+  /// Callback which is called to set the painter of the given [FlSpot].
+  /// The [FlSpot] is provided as parameter to this callback
+  final GetDotPainterCallback getDotPainter;
+
   /// set [show] false to prevent dots from drawing,
-  /// [dotSize] determines the size of dots.
   /// if you want to show or hide dots in some spots,
   /// override [checkToShowDot] to handle it in your way.
-  ///
-  /// The color of dot determines by [getDotColor],
-  /// you can override it, it gives you a [FlSpot],
-  /// and you should decide to return a [Color].
-  ///
-  /// You can have stroke line around the dot,
-  /// by setting the thickness by [strokeWidth],
-  /// and you can implement [getStrokeColor] callback,
-  /// it gives you the [FlSpot], and you should decide to return a [Color].
   FlDotData({
     bool show,
-    double dotSize,
     CheckToShowDot checkToShowDot,
-    double strokeWidth,
-    GetDotColorCallback getStrokeColor,
-    GetDotColorCallback getDotColor,
+    GetDotPainterCallback getDotPainter,
   })  : show = show ?? true,
-        dotSize = dotSize ?? 4.0,
         checkToShowDot = checkToShowDot ?? showAllDots,
-        strokeWidth = strokeWidth ?? 0.0,
-        getStrokeColor = getStrokeColor ?? _defaultGetDotStrokeColor,
-        getDotColor = getDotColor ?? _defaultGetDotColor;
+        getDotPainter = getDotPainter ?? _defaultGetDotPainter;
 
   /// Lerps a [FlDotData] based on [t] value, check [Tween.lerp].
   static FlDotData lerp(FlDotData a, FlDotData b, double t) {
     return FlDotData(
       show: b.show,
       checkToShowDot: b.checkToShowDot,
-      dotSize: lerpDouble(a.dotSize, b.dotSize, t),
-      strokeWidth: lerpDouble(a.strokeWidth, b.strokeWidth, t),
-      getDotColor: b.getDotColor,
-      getStrokeColor: b.getStrokeColor,
+      getDotPainter: b.getDotPainter,
     );
   }
 
   /// Used for equality check, see [EquatableMixin].
   @override
   List<Object> get props => [
         show,
-        dotSize,
-        strokeWidth,
-        getStrokeColor,
-        getDotColor,
         checkToShowDot,
+        getDotPainter,
+      ];
+}
+
+abstract class FlDotPainter with EquatableMixin {
+  void draw(Canvas canvas, FlSpot spot, Offset offsetInCanvas);
+
+  Size getSize(FlSpot spot);
+}
+
+class FlDotCirclePainter extends FlDotPainter {
+  Color color;
+  double radius;
+  Color strokeColor;
+  double strokeWidth;
+
+  FlDotCirclePainter({
+    Color color,
+    double radius,
+    Color strokeColor,
+    double strokeWidth,
+  })  : color = color ?? Colors.green,
+        radius = radius ?? 4.0,
+        strokeColor = strokeColor ?? Colors.green,
```

Comment:
please make a differentiate between `color`, and `strokeColor`, I prefer `Colors.green.darken()` for strokeColor.